rust/rpmostree-client/src/lib.rs

```diff
@@ -0,0 +1,66 @@
+//! APIs for interacting with rpm-ostree client side.
+
+use anyhow::Context;
+use serde_derive::Deserialize;
+use std::process::Command;
+
+/// Our generic catchall fatal error, expected to be converted
+/// to a string to output to a terminal or logs.
+type Result<T> = std::result::Result<T, Box<dyn std::error::Error + Send + Sync + 'static>>;
```

Comment:
Can we use thiserror instead? (Obviously can be a follow-up.)

Reply:
I should write this one up too but...personally I try hard to avoid introducing custom error types like that until it's *truly* required.  Or to say it another way the `Result<>` is intended to be used basically as just a string - treat it as fatal and don't try to introspect it.

The classic example here being rather than trying to catch "enoent" from status to detect if rpm-ostree is installed, we have a separate `is_rpmostree_installed() -> Result<bool>` method.